### PR TITLE
Starting "Box2D - TextBed" adaption (most demos working)

### PR DESCRIPTION
**Pull request by aismann**

### CODE REVIEW

**Comment by aismann** [comment]
halx99:  Can you have a look on the linux build, ply?
It should build for build-win32 (all other should be the changes ignoring and not using)

**Comment by halx99** [comment]
I mark this purll-request as `draft`, if you done everythings, please inform me, or click the button `Ready for review`

File: tests/cpp-tests/Classes/controller.cpp [hunk 1]
```diff
@@ -56,11 +56,11 @@ class RootTests : public TestList
 
 
         addTest("Box2D - Basic", []() { return new (std::nothrow) Box2DTests(); });
-#if defined(CC_PLATFORM_PC)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32)
```
Comment (halx99) [inline tests/cpp-tests/Classes/controller.cpp:59]: The box2d testbed can't be run on linux or mac ?

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: I have No system to chack it.
IT should

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: Chipmunk testbed also

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: ok, I will check them

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: The prob is:
I will add Mouse and Keyboard Support later and after all i will move Dome Files into extension

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: I belive some thirt Party stuff have to be adapt dir Linux/Mac

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: For cross-platform adaption, you can refer to https://github.com/halx99/adxe/commit/3d7251e7e3864713dd2184da62779bcdfcd9be86

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: You will pull all changes to the dev branch?
Or have I to somthing before?

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: If the changes no conflicit with your branch `Box2D-TestBed`, I can create a pr with my branch

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: Thanks.

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: Checked, no confilcit, and I create a pull-request: https://github.com/adxeproject/adxe/pull/420

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: Perfect. Thanks!
I will working again after all checkins.

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: The pr 420 was merged, can I merge this pr before you continue working?

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: That what I mean.
I will wait.

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: Merged, you can create a new branch from upstream:dev and continue working

**Comment by aismann** [comment]
Is Linux and iOS working with both TestBeds now?

**Comment by halx99** [comment]
Shoud be work, you can try, if there are any issues, I can solve with you

**Comment by aismann** [comment]
I have no Linux/IOS system her, i cannot try it.
(What I mean is runtime errors)

**Comment by halx99** [comment]
Normally, everythings should be works on both platforms supported by `adxe`, I can try them on linux/ios.
And also, If there is no bad compile warnings, the program should be works fine on all platforms

File: tests/cpp-tests/Classes/controller.cpp (hunk 1 above)
Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: > 
> 
> The box2d testbed can't be run on linux or mac ?

Have you tested it on mac?

I see on the last upload on controller.cpp:   (So only Windows can the the menu "Box2D - TestBed")
```
#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32)
        addTest("Box2D - TestBed", []() { return new (std::nothrow) Box2DTestBedTests(); });
#endif
```

Comment (halx99) [reply tests/cpp-tests/Classes/controller.cpp:59]: I run the chipmunk-testbed on mac, works fine.

Comment (aismann) [reply tests/cpp-tests/Classes/controller.cpp:59]: Box2D - Testbed is crashing on linux

**Pull request merged**